src/components/FeatureViews/AdminPortal/AdminUtils.js

```diff
@@ -0,0 +1,173 @@
+import { isSunday } from "date-fns";
+
+export const roleColor = (string)=>{
+    switch(string){
+        case "administrator":
+            return "red";
+        case "parent":
+            return "#812CD7";
+        case "student":
+            return "#378BFF";
+        case "instructor":
+            return "#1FA18A"
+        case "receptionist":
+            return "#D04E3D"
+        default:
+            return "no match"
+    }
+}
+
+export const statusColor = (x)=> {
+    switch(x){
+        case "0":
+            return "red";
+        case "1":
+            return "red";
+        case "2":
+            return "yellow";
+        case "3":
+            return "yellow";
+    }
+}
```

Comment:
This can be simplified to an object like so:

    export const statusColor = {
        "0": "red",
        "1": "red",
        "2": "yellow",
        "3": "yellow",
    };

(note that you will then have to use it as `statusColor[status]` instead of with parentheses)

If you are checking for it being in a range, use an `if` statement instead.

If the keys are only going to be integers starting at 0, then you can alternatively use an array:

    export const statusColor = ["red", "red", "yellow", "yellow"];